src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -42,6 +84,15 @@ public static List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinitio
                         continue;
                     }
 
+                    // Skip get_IsCompleted to avoid unuseful branch due to async/await state machine
+                    if (isAsyncStateMachineMoveNext && instruction.Previous.Operand is MethodReference operand &&
+                            operand.Name == "get_IsCompleted" &&
+                            operand.DeclaringType.FullName.StartsWith("System.Runtime.CompilerServices.TaskAwaiter") &&
```

Comment:
Typesafe gets a bit more complicated here since it would have to look at both `TaskAwaiter` and handle the generic `TaskAwaiter<TResult>`, but I think that's more future-proof than a `StartsWith()`.

Reply:
Same as above we have to work with `Mono.Cecil.TypeReference` metadata and not with `System.Type`